var/spack/repos/builtin/packages/xolotl/package.py

```diff
@@ -0,0 +1,64 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Xolotl(CMakePackage):
+    """Xolotl is a high-performance computing code using
+    advection-reaction-diffusion (ADR) kinetic rate theory to model the
+    time evolution of the divertor material in next generation tokamaks,
+    like ITER, as well as nuclear fuel in fission reactors."""
+
+    homepage = "https://github.com/ORNL-Fusion/xolotl"
+    url = "https://github.com/ORNL-Fusion/xolotl/archive/refs/tags/v3.1.0.tar.gz"
+    git = "https://github.com/ORNL-Fusion/xolotl.git"
+
+    # notify when the package is updated.
+    maintainers("sblondel", "PhilipFackler")
+
+    license("BSD-3-Clause", checked_by="PhilipFackler")
+
+    version("3.1.0", sha256="68a495ab0c3efb495189f73474d218eb591099b90d52d427ac868b63e8fc2ee8")
+
+    depends_on("cxx", type="build")
+
+    depends_on("mpi")
+    depends_on("boost +log +program_options")
+    depends_on("hdf5 +mpi")
+
+    variant("int64", default=True, description="Use 64-bit indices")
+    variant("cuda", default=False, description="Activates CUDA backend")
```

Comment:
Prefer to inherit from CudaPackage and then there is no need to define your own variant.